### PR TITLE
add status.labelSelector field to unitedDeployment to support scale sub-resource

### LINKED ISSUE #1308: [feature request] UnitedDeployment supports scale sub-resource
**What would you like to be added**:
add `status.selector` field to unitedDeployment to support scale sub-resource

**Why is this needed**:
support HPA

### PR DESCRIPTION
**Pull request by diannaowa**
### Ⅰ. Describe what this PR does
Add status.labelSelector field to unitedDeployment to support scale sub-resource

### Ⅱ. Does this pull request fix one issue?

fixes #1308

### CODE REVIEW

**Comment by diannaowa** [comment]
This is a romantic PR id

File: pkg/controller/uniteddeployment/uniteddeployment_controller.go [hunk 1]
```diff
@@ -231,6 +233,14 @@ func (r *ReconcileUnitedDeployment) Reconcile(_ context.Context, request reconci
 		return reconcile.Result{}, err
 	}
 
+	selector, err := util.ValidatedLabelSelectorAsSelector(instance.Spec.Selector)
```
Comment (furykerry) [inline pkg/controller/uniteddeployment/uniteddeployment_controller.go:236]: just using ValidatedLabelSelectorAsSelector is not enough to identify some label selector errors, you can call LabelSelectorAsSelector and check if there is any error. Similar code is in the cloneset validation validateCloneSetSpec

Comment (diannaowa) [reply pkg/controller/uniteddeployment/uniteddeployment_controller.go:236]: Do you mean validate UnitedDeploymentSpec? https://github.com/openkruise/kruise/blob/9e74506ca800877f5bf63723efa33521a3533c46/pkg/webhook/uniteddeployment/validating/uniteddeployment_validation.go#LL52C60-L52C60

Comment (furykerry) [reply pkg/controller/uniteddeployment/uniteddeployment_controller.go:236]: yes, it seems the uniteddeployment webhook already validate the selector, is it necessary to validate again the controller

Comment (diannaowa) [reply pkg/controller/uniteddeployment/uniteddeployment_controller.go:236]: LabelSelectorAsSelector has been called in the validateUnitedDeploymentSpec method to identify some label selector errors.
I think ValidatedLabelSelectorAsSelector is ok here.

Comment (diannaowa) [reply pkg/controller/uniteddeployment/uniteddeployment_controller.go:236]: > yes, it seems the uniteddeployment webhook already validate the selector, is it necessary to validate again the controller

Unnecessary, i think.

**Review by furykerry — COMMENTED** [review]
/lgtm

**Pull request merged**